openeo_processes_dask/process_implementations/comparison.py

```diff
@@ -0,0 +1,127 @@
+from typing import Callable, Optional
+
+import dask.array as da
+import numpy as np
+import xarray as xr
+from numpy.typing import ArrayLike
+from xarray.core.duck_array_ops import notnull
+
+from openeo_processes_dask.process_implementations.utils import check_type, is_number
+
+__all__ = [
+    "is_infinite",
+    "is_valid",
+    "eq",
+    "neq",
+    "gt",
+    "gte",
+    "lt",
+    "lte",
+    "between",
+]
+
+
+def is_infinite(x: ArrayLike):
+    if x is None:
+        return False
+    if check_type(x, "str") or check_type(x, "list") or check_type(x, "dict"):
+        return False
+    return np.isinf(x)
+
+
+def is_valid(x: ArrayLike):
+    finite = np.logical_not(is_infinite(x))
+    return np.logical_and(notnull(x), finite)
+
+
+def eq(
+    x: ArrayLike,
+    y: ArrayLike,
+    delta: Optional[float] = None,
+    case_sensitive: Optional[bool] = True,
+):
+    if is_number(x, y):
+        if not is_valid(x) or not is_valid(y):
+            return np.nan
+        if (
+            isinstance(x, bool)
+            and not isinstance(y, bool)
+            or isinstance(y, bool)
+            and not isinstance(x, bool)
+        ):
+            return False
```

Comment:
https://github.com/Open-EO/openeo-processes-dask/pull/50#discussion_r1193779887

I now added this check to make sure that the example of  eq(x = 0, y = false) => false works, but there is one problem left: 
I do this check for inputs that only consist of one value, e.g. int, float, bool. As soon as I want to do this for an array and check if it is a bool array, it might not work, because numpy easily converts a bool array into a float array. In that case, False would be set to 0 and the example would not work.

Reply:
@LukeWeidenwalker I tried implementing the eq(x = 0, y = false) => false example with an explicit check for the bool type if the input value only consists of one number - as soon as there are numpy or dask arrays, I would not check it explicitely, is this fine for you? or should I add a check for the numpy array dtype?